src/plugins/saml-auth/components/AuthorizationPopup.jsx

```diff
@@ -0,0 +1,155 @@
+import React from "react"
+import PropTypes from "prop-types"
+import DefinitionSelect from "./DefinitionSelect"
+import { SAML_AUTH_STATE_LOGGED_IN, SAML_AUTH_STATE_LOGGING_IN, SAML_AUTH_STATE_FAILED } from "../actions"
+
+export default class AuthorizationPopup extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedDefinitionOption: null,
+    }
+  }
+
+  close = () => {
+    const { authActions, errActions } = this.props
+
+    errActions.clear({ type: "auth" })
+    authActions.showDefinitions(false)
+  };
+
+  onSelectDefinition = (definition) => {
+    const { errActions } = this.props
+
+    errActions.clear({ type: "auth" })
+    this.setState({ selectedDefinitionOption: definition })
+  };
+
+  componentDidUpdate() {
+    const { samlAuthSelectors } = this.props
+    const isSamlAuthenticated =
+      samlAuthSelectors.samlAuthState() ===
+      SAML_AUTH_STATE_LOGGED_IN
+    const isSamlFailed = samlAuthSelectors.samlAuthState() === SAML_AUTH_STATE_FAILED
+    if (isSamlAuthenticated) {
+      // unset url search
+      this.close()
+      window.history.pushState({}, document.title, window.location.pathname)
+    } else if(isSamlFailed) {
+      window.history.pushState({}, document.title, window.location.pathname)
+    }
+  }
+
+  render() {
+    let {
+      samlAuthSelectors,
+      authSelectors,
+      authActions,
+      getComponent,
+      errSelectors,
+      specSelectors,
+      getSystem,
+      fn: { AST = {} },
+    } = this.props
+    let definitions = authSelectors.shownDefinitions()
+    let authorized = authSelectors.authorized()
+
+    let Auths = getComponent("auths")
+    let loginDisclaimer = specSelectors.spec().get("loginDisclaimer")
+
+    let { selectedDefinitionOption } = this.state
+    let errors = errSelectors
+      .allErrors()
+      .filter((err) => err.get("authId") === "SamlAuth")
+    let hasErrors = errors.size > 0
+    let isAuthenticated = authorized.size > 0
+    let authenticatedKey = authorized.keySeq().first()
+    let selectedDefinitionKey = isAuthenticated ? authenticatedKey : selectedDefinitionOption
+
+    let isSamlAuthenticating =
+      samlAuthSelectors.samlAuthState() === SAML_AUTH_STATE_LOGGING_IN
+    let showLoginOptions = !isSamlAuthenticating && !selectedDefinitionKey
+    let showLoginStep =
```

Comment:
rename to `showLoginAuth` instead? It returns `<Auths>`. `showLoginStep` is unclear